scriptmodules/supplementary/runcommand/runcommand.sh

```diff
@@ -278,33 +278,54 @@ function get_all_x11_modes()
 {
     declare -Ag MODE
     local id
-    local info
     local line
-    local verbose_info=()
-    local output="$($XRANDR --verbose | grep " connected" | awk '{ print $1 }')"
-
-    while read -r line; do
-        # scan for line that contains bracketed mode id
-        id="$(echo "$line" | awk '{ print $2 }' | grep -o "(0x[a-f0-9]\{1,\})")"
-
-        if [[ -n "$id" ]]; then
-            # strip brackets from mode id
-            id="$(echo ${id:1:-1})"
-
-            # extract extended details
-            verbose_info=($(echo "$line" | awk '{ for (i=3; i<=NF; ++i) print $i }'))
-
-            # extract x/y resolution, vertical refresh rate and append details
-            read -r line
-            info="$(echo "$line" | awk '{ print $3 }')"
-            read -r line
-            info+="x$(echo "$line" | awk '{ print $3 }') @ $(echo "$line" | awk '{ print $NF }') ("${verbose_info[*]}")"
-
-            # populate resolution into arrays
-            MODE_ID+=($output:$id)
-            MODE[$output:$id]="$info"
-        fi
-    done < <($XRANDR --verbose)
+    while read -r id; do
+        # populate CONNECTOR:0xID into an array
+        MODE_ID+=($id) # output:id as in (hdmi:0x44)
+        
+        read -r line
+        # array is x/y resolution @ vertical refresh rate ( details )
+        MODE[$id]="$line"
+    done < <( $XRANDR --verbose | awk '
+        # defines the type of line
+        # true is the "header" (output and id)
+        # false is the "description" (Mode: and everything that begins with a space)
+        { type = /^[^ \t]+/ }
+        
+        # Exit after the first output
+        type && output {exit} # New header and output set means new output
+        #type { output="" } # New header, Reset output
```

Comment:
Is this commented out line supposed to be left?

Reply:
Yes. Everything works fine by not exiting. but because get_x11_mode_info exits after the first output found, I thought it best to also do the same for now. if you remove the 2 lines above and uncomment you would really get all the x11 modes. but it might break with x11_mode_info. So I want to keep that commented until that ever gets fixed.